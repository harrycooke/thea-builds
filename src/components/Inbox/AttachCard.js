import React from 'react';
import axios from 'axios';
import {
  Card,
  CardActions,
  Modal,
  Fab,
  MobileStepper,
  Button,
  Link
} from "@material-ui/core";
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import IconButton from '@material-ui/core/IconButton';
import GetAppIcon from '@material-ui/icons/GetApp';
import { connect } from "react-redux";
import CloseIcon from "@material-ui/icons/Close";
import { withStyles } from '@material-ui/styles';
import { Document, Page, pdfjs } from "react-pdf";
import { uploadFile } from '../../utils/s3';
import { getServerUrl } from '../../utils/functions';
import styles from "../../assets/styles/AttachCardStyles";
pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

export class AttachCard extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      attachName: this.props.attachName,
      displayModal: false,
      numPages: null,
      pageNumber: 1,
      activeStep:0,
      imageStatus: true
    }
  }

  onDocumentLoadSuccess = ({ numPages }) => {
    this.setState({ numPages });
    // console.log("Got here: ", numPages);
  }

  handleImageLoaded = () => {
    this.setState({ imageStatus: false });
  }
  
  handleCardClick = (fileIndex) => {
    this.setState({
      displayModal: true,
      activeStep: fileIndex
    })
  }

  handleClose = () => {
    this.setState({
      displayModal: false
    })
  }

  uploadFile = async () => {
    this.setState({ isLoading: true });
    const { oldFileList, filePaths } = this.state;
    const promise = await Promise.all(oldFileList.map(async (file) => {
      return await uploadFile(file);
    }));
    promise.map(path => filePaths.push(path));
    axios.post(getServerUrl() + '/api/v1/case/getFileLink', filePaths)
      .then(res => {
        this.setState({
          uploadedLink: res.data.data,
          isLoading: false,
          oldFileList:[]
        })
      })
      .catch(err => {
        this.setState({ isLoading: false })
      })
  }

  handleNext = () => {
    this.setState({activeStep : this.state.activeStep + 1});
  };

  handleBack = () => {
    this.setState({activeStep : this.state.activeStep - 1});
  };

  pdfNext = () => {
    this.setState({pageNumber : this.state.pageNumber + 1});
  }
  pdfPrev = () => {
    this.setState({pageNumber : this.state.pageNumber - 1});
  }

  downloadFile = (url) => {
    fetch(url, {
      method: "GET",
      headers: {}
    })
    .then(response => {
      response.arrayBuffer().then(function(buffer) {
        let ext = response.url.split('.').pop().split(/\#|\?/)[0];
        let fileName = 'file_' + new Date().getTime() + '.' + ext;
        const url = window.URL.createObjectURL(new Blob([buffer]));
        const link = document.createElement("a");
        link.href = url;
        link.setAttribute("download", fileName);
        document.body.appendChild(link);
        link.click();
        link.remove();
      });
    })
    .catch(err => {
      console.log(err);
    });
  }

  render() {
    const { uploadedLink, displayModal, pageNumber, numPages, activeStep, files } = this.state;
    const { classes, fileObj, currentCase, fileIndex } = this.props;
    const caseFiles = currentCase.caseData ? currentCase.caseData.files : [];
    const maxSteps = caseFiles.length;

    let filesUploaded;
    if (files && files.length >= 1) {
      filesUploaded = files.map((file, i) => (
          <img src={this.state.fileUrls[i]} alt={this.state.fileName[i]} style={{ width: "auto", height: "24vh", backgroundColor: "grey" }}  />
      ));
    } else {
      filesUploaded = []
    }


    let fileDisplay;
    let fileType = fileObj.type.substring(0, fileObj.type.indexOf("/"));
    let fileExt = fileObj.type.split('/').pop();
    
    if (fileType === 'image') {
      fileDisplay = <>{this.state.imageStatus && "Loading image"}<img src={fileObj.name} onLoad={()=>this.handleImageLoaded(this)} alt="Uploaded image" style={{ width: "auto", height: "24vh" }} /> 
      </>
    }
    else if (fileType === 'application' && fileExt === 'pdf') {
      fileDisplay = <div><div><Document
        file={fileObj.name} alt="Uploaded document"
        onLoadSuccess={this.onDocumentLoadSuccess.bind(this)} style={{ width: "auto", height: "24vh" }}
        >
        <Page pageNumber={pageNumber} />
      </Document></div>
      </div>
    }
    else if (fileType === 'application' && fileExt !== 'pdf') {
      fileDisplay = <p style={{ width: "70%", alignItems: "center", color: "grey" }}><b>Download to view</b><br/> {fileExt} file  </p>
    }
    else if (fileType === 'video') {
      fileDisplay = <><video src={fileObj.name} alt="Uploaded video" controls style={{ width: "auto", height: "24vh" }} /> 
      </>                 
    }
    else if (fileType === 'text') {                         
      fileDisplay = <p style={{ width: "70%", alignItems: "center", color: "grey" }}><b>Download to view</b><br/> {fileExt} file  </p>
    }
    else {                         //Not required this elseif but added as a safety net for any filetype
      fileDisplay = <p style={{ width: "70%", alignItems: "center", color: "grey" }}><b>Download to view</b><br/> {fileExt} file  </p>
    }



    /*===================*/
    let viewFiles = '';
    let fileTypeData = caseFiles[activeStep].type;
    let fileTypeRes = fileTypeData.substring(0, fileTypeData.indexOf("/"));
    let fileExtRes = fileTypeData.split('/').pop();
    
    if(fileTypeRes === 'image'){
      viewFiles = <>{this.state.imageStatus && "Loading file"}<img
                className={classes.img}
                onLoad={()=>this.handleImageLoaded(this)}
                src={caseFiles[activeStep].name}
                alt={caseFiles[activeStep].type}
              /></>
    }
    else if(fileTypeRes === 'application' && fileExtRes === 'pdf'){
      viewFiles = <Document
                    file={caseFiles[activeStep].name}
                    onLoadSuccess={this.onDocumentLoadSuccess.bind(this)}
                    style={{ width: "100%", height: "100%" }}
                  >
                    <p className={classes.textAlign}>
                      <Button size="small" className={classes.floatLeft} onClick={this.pdfPrev} disabled={pageNumber === 1}>
                        {styles.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
                        Prev page
                      </Button>
                      Page {pageNumber} of {numPages}
                      <Button size="small" className={classes.floatRight} onClick={this.pdfNext} disabled={pageNumber === numPages}>
                        Next page
                        {styles.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
                      </Button>
                    </p>
                    <Page pageNumber={pageNumber} className={classes.pdfView}/>
                  </Document>
    }
    else if(fileTypeRes === 'application' && fileExtRes !== 'pdf'){
      viewFiles = <h1 style={{ width: "70%", alignItems: "", color: "grey", marginTop:"20%" }}>{fileExtRes} file </h1>
    }
    else if (fileTypeRes === 'video') {
      viewFiles = <video src={caseFiles[activeStep].name} controls style={{ width: "100%", height: "100%" }} />
    }
    else if (fileTypeRes === 'text') {
      viewFiles = <h1 style={{ width: "70%", alignItems: "", color: "grey", marginTop:"30vh" }}>{fileExtRes} file</h1>
    }
    else {
      viewFiles = <h1 style={{ width: "70%", alignItems: "", color: "grey", marginTop:"30vh" }}>{fileExtRes} file</h1>
    }
     
    
    return (
      <div>
        <div className={classes.container}>
          <Card className={classes.contentCard} >
            <CardActions
              className={classes.cardAction} 
              onClick={event => this.handleCardClick(fileIndex) }
              // key={fileObj.type} style={ width= {fileObj.type === 'application' ? "24vh" : "auto" }}
              >
              {fileDisplay}
            </CardActions>

            <div>
              {/* {files && files.length >= 1 && (
                  filesUploaded = files.map((file, i) => (
                    <p>{file.name.slice(13)}Case</p>                                 //Slices the initial substring of Epoch time (13 char) from the filename
                ))
              )} */}

              <p>
                {/*<Link href={fileObj.name} download target="_blank" style={{ float: "right" }} >*/}
                <Link onClick={() => this.downloadFile(fileObj.name)} style={{ float: "right" }} >
                  {/* {console.log("this is the file" + fileObj.name)} */}
                  <IconButton >
                    <GetAppIcon style={{ marginLeft: 5, color:"#004866" }} className={classes.title} />
                  </IconButton>
                </Link>
              </p>
            </div>
          </Card>
        </div>

        <Modal
          aria-labelledby="simple-modal-title"
          aria-describedby="simple-modal-description"
          open={displayModal}
          onClose={this.handleClose}
          BackdropProps={{ style: { top: 64, left: "0%" } }}                //The grey Modal background when a file is opened
        >
          <div className={classes.modalContainer}>
            <Card className={classes.modalCard} style={{ borderBottomRightRadius: 0, borderBottomLeftRadius: 0 }}>
              {viewFiles}
            </Card>

            <MobileStepper
              steps={maxSteps}
              position="static"
              variant="text"
              activeStep={activeStep}
              nextButton={
                <Button size="small" onClick={this.handleNext} disabled={activeStep === maxSteps - 1} style={{ color:"#fff" }}>
                  Next
                  {styles.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
                </Button>
              }
              backButton={
                <Button size="small" onClick={this.handleBack} disabled={activeStep === 0} style={{ color:"#fff" }}>
                  {styles.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
                  Back
                </Button>
              }
            />
            <div style={{
              justifyContent: "center",
              display: "flex",
              paddingTop: 5
            }}>
              <Fab
                variant="round"
                className={classes.closeButton}
                onClick={this.handleClose}>
                <CloseIcon />
              </Fab>
            </div>
          </div>
        </Modal>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  currentCase: state.case.currentCase
});


export default connect(
  mapStateToProps,{}
)(withStyles(styles)(AttachCard));