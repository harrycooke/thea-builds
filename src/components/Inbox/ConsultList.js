import React from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { makeStyles } from '@material-ui/core/styles';
import _ from "../../@lodash";
import {
  Table,
  TableBody,
  TableCell,
  TablePagination,
  TableRow,
  TableHead,
  Paper,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Slide  
} from "@material-ui/core";
import { fetchUserData } from "../../actions/authActions";
import { modifyCase } from "../../actions/caseAction";

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

function desc(a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

function stableSort(array, cmp) {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = cmp(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  return stabilizedThis.map(el => el[0]);
}

function getSorting(order, orderBy) {
  return order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy);
}

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(3),
  },
  paper: {
    width: '100%',
    marginBottom: theme.spacing(2),
    boxShadow: "none",
  },
  table: {
    // minWidth: 750,
  },
  tableWrapper: {
    overflowX: 'auto',
  },
  tableCellBold: {
    color: "#1C1C1C",
    fontSize: 20,
    fontWeight: 500,
  },
  tableCellLight: {
    color: "#1c1c1c",
    fontSize: 18,
    fontWeight: "normal"
  }
}));

//This is not that imp in terms of testing

export function ConsultList(props) {
  const classes = useStyles();
  const [order] = React.useState('desc');
  const [orderBy] = React.useState('id');
  const [page, setPage] = React.useState(0);
  const [dense] = React.useState(false);
  const [rowsPerPage, setRowsPerPage] = React.useState(10);
  const [selectedConsult, setSelected] = React.useState(props.selectedConsult);
  const [selectedtemp, setSelectedtemp] = React.useState(null);
  const [emailTo, setEmailTo] = React.useState(null);

  let rows = [], rowsLength = 0;
  if (props.rows) {
    rows = props.rows;
    rowsLength = rows.length;
  }
  //Confirm Dialog 
  const [open, setOpen] = React.useState(false);
  const [unassignmodal, setUnassignmodal] = React.useState(false);

  const handleConfirmDialog = (e, id, emailOfSpecialist) => {
    setSelectedtemp(id);
    setEmailTo(emailOfSpecialist);
    setOpen(true);
  };

  const handleClose = () => {
    setEmailTo(null);
    setOpen(false);
  };

  const assignConsultant = (e, consultId, emailTo, caseData) => {
    if(consultId){
      const { case_id } = props.params;
      const current_time = new Date();
      const current_timestamp = current_time.getTime();
      const newCaseData = {
        caseId: case_id,
        last_updated_by_id: props.userdata.id,
        updated_at: current_timestamp,
        specialist_id: consultId,
        emailTo: emailTo
      };
      let readByOthers = caseData.is_read_by_other ? caseData.is_read_by_other : [];
      let ifIndex = _.indexOf(readByOthers, props.userdata.id);
      if(ifIndex < 0){
        readByOthers.push(props.userdata.id);
        newCaseData['is_read_by_other'] = readByOthers;
      }
      props.modifyCase(newCaseData);
      setSelected(consultId);
      setEmailTo(null);
      setOpen(false);
    }
  };

  const handleUnassignDialog = (e, id) => {
    setSelectedtemp(id);
    setUnassignmodal(true);
    setOpen(true);
  };

  const unassignConsultant = (e, consultId ) => {
    if(consultId === 0){
      const { case_id } = props.params;
      const current_time = new Date();
      const current_timestamp = current_time.getTime();
      const newCaseData = {
        caseId: case_id,
        last_updated_by_id: props.userdata.id,
        updated_at: current_timestamp,
        specialist_id: consultId
      };
      props.modifyCase(newCaseData);
      setSelected(null);
      setUnassignmodal(false);
      setOpen(false);
    }
  }  

  const handleUnassignClose = () => {
    setUnassignmodal(false);
    setOpen(false);
  };

  function handleChangePage(event, newPage) {
    setPage(newPage);
  }

  function handleChangeRowsPerPage(event) {
    setRowsPerPage(+event.target.value);
  }

  return (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <div className={classes.tableWrapper}>
          <Table
            className={classes.table}
            aria-labelledby="tableTitle"
            size={dense ? 'small' : 'medium'}
          >
            <TableHead>
              <TableRow>
                <TableCell padding="checkbox" align="center">Select Specialist</TableCell>
                <TableCell align="center">First Name</TableCell>
                <TableCell align="center">Last Name</TableCell>
                <TableCell align="center">Specialty</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {stableSort(rows, getSorting(order, orderBy))
                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                .map((row, index) => {
                  return (

                    <TableRow
                      hover
                      selected={selectedConsult === row.id ? true: false}
                      tabIndex={-1}
                      key={index}
                    >
                      <TableCell align="center" padding="checkbox">
                        <div>
                        {selectedConsult === row.id ?
                          <Button variant="contained" color="primary" onClick={event => handleUnassignDialog(event, 0)}>Unassign</Button>
                          :
                          <Button variant="outlined" onClick={event => handleConfirmDialog(event, row.id, row.email)}>Assign</Button>
                        }
                        </div>
                      </TableCell>

                      <TableCell align="center" className={classes.tableCellBold}>
                        {row.firstName ? row.firstName : null}
                      </TableCell>

                      
                      <TableCell
                        align="center"
                        className={classes.tableCellBold}
                      >
                        {row.lastName ? row.lastName : null}
                      </TableCell>
                      
                      <TableCell align="center"  className={classes.tableCellBold}>
                      {row.specialty ? row.specialty : null}
                      </TableCell>
                    </TableRow>
                  );
                })}

              {rowsLength === 0 && (
                <TableRow style={{ height: 49 }}>
                  <TableCell colSpan={6}> No Consult.. </TableCell>
                </TableRow>
              )}
            </TableBody>
          </Table>
          <Dialog
            open={open}
            TransitionComponent={Transition}
            keepMounted
            onClose={unassignmodal ? handleUnassignClose: handleClose }
            aria-labelledby="alert-dialog-slide-title"
            aria-describedby="alert-dialog-slide-description"
          >
            <DialogTitle id="alert-dialog-slide-title">{"Please confirm"}</DialogTitle>
            <DialogContent>
              <DialogContentText id="alert-dialog-slide-description">
                Are you sure you want to {unassignmodal ?  'unassign': 'assign'} this specialist {unassignmodal ?  'from': 'to'} {props.patientName}?
              </DialogContentText>
            </DialogContent>
            {unassignmodal ?
              <DialogActions>
                <Button onClick={handleUnassignClose} color="primary">
                  No
                </Button>
                <Button onClick={event => unassignConsultant(event, selectedtemp)} color="primary">
                  Yes
                </Button>
              </DialogActions>
            :
              <DialogActions>
                <Button onClick={handleClose} color="primary">
                  No
                </Button>
                <Button onClick={event => assignConsultant(event, selectedtemp, emailTo, props.currentCaseData)} color="primary">
                  Yes
                </Button>
              </DialogActions>
            }
          </Dialog>
        </div>
        <TablePagination
          rowsPerPageOptions={[5, 10, 15]}
          component="div"
          count={rows.length}
          rowsPerPage={rowsPerPage}
          page={page}
          backIconButtonProps={{
            'aria-label': 'Previous Page',
          }}
          nextIconButtonProps={{
            'aria-label': 'Next Page',
          }}
          onChangePage={handleChangePage}
          onChangeRowsPerPage={handleChangeRowsPerPage}
        />
      </Paper>
    </div>
  );
}

ConsultList.propsTypes = {
  fetchUserData: PropTypes.func.isRequired,
  userdata: PropTypes.object.isRequired,
  modifyCase: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata
})

export default connect(
  mapStateToProps,
  { fetchUserData, modifyCase }
)(ConsultList)
