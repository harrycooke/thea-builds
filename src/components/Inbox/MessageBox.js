import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({

});

export class MessageBox extends Component {
  render() {
    return (
      <div></div>
    );
  }
}

export default withStyles(styles)(MessageBox);