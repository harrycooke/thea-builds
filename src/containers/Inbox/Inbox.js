import React, { Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar';
import {
  Card,
  Grid,
  Typography,
  Button,
  Hidden,
  MenuItem,
  TextField,
  Divider,
  InputBase,
  InputAdornment,
  Snackbar,
} from '@material-ui/core';
import Media from 'react-media';
import SearchIcon from '@material-ui/icons/Search';
import AddIcon from "@material-ui/icons/Add";
import CreateIcon from '@material-ui/icons/Create';

import _ from "../../@lodash";
import { fetchAllCases, setCurrentCase, modifyCase, setSnackbar } from "../../actions/caseAction";
import { fetchAllDrafts, setCurrentDraft, removeDraft } from "../../actions/draftActions";
import MessageList from "../../components/Inbox/MessageList";
import styles from "../../assets/styles/InboxStyles";
import { specialtyList } from "../../utils/constant";
import ConfirmationDialog from "../../components/SharedComponents/ConfirmationDialog"
import { removeFileFromS3 } from '../../utils/s3';
import CircularProgress from '../../components/SharedComponents/CircularProgress';

import { sendAmplitudeData } from '../../utils/amplitude';

const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

class Inbox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      redirectToChat: false,
      redirectToAssignConsult: false,
      cases_id: null,
      specialty: null,
      redirectToCreateCase: false,
      readAllCases: [],
      readAllDrafts: [],
      selectedSpecialty: '',
      searchResultsForCases: [],
      searchResultsForDrafts: [],
      confirmDraftDelete: false,
      files: []
    }
    this.removeDraft = this.removeDraft.bind(this)
    this.openDraftDialog = this.openDraftDialog.bind(this)
    this.handleClose = this.handleClose.bind(this)
  }

  handleTrClicked = (id, isDraft, caseData) => {
    if (isDraft) {
      this.setState({
        redirectToCreateCase: true,
        selectedClientID: id
      });
      this.props.setCurrentDraft({ case_id: id })
    } else {
      let newCaseData = {
        caseId: id,
        last_updated_by_id: this.props.userdata.id
      }
      if(this.props.userdata.id === caseData.created_by_id){
        newCaseData['is_read'] = true;
      }else{
        let readByOthers = caseData.is_read_by_other ? caseData.is_read_by_other : [];
        let ifIndex = _.indexOf(readByOthers, this.props.userdata.id);
        if(ifIndex < 0){
          readByOthers.push(this.props.userdata.id);
          newCaseData['is_read_by_other'] = readByOthers;
        }
      }
      this.props.modifyCase(newCaseData);
      this.setState({
        redirectToChat: true,
        selectedClientID: id
      });
    }
  }

  handleSearch = e => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
    let selectedSpecialty = e.target.value;
    const { readAllCases, readAllDrafts } = this.state;
    if (selectedSpecialty !== '') {
      let searchResultsForCases = readAllCases;
      let searchResultsForDrafts = readAllDrafts;
      // debugger;
      if(selectedSpecialty != "All Specialties"){                                         //Need to bug-proof here. If the text value to if condition is changed, the condition will break and no value returned.
        searchResultsForCases = _.filter(readAllCases, function (item) {
          return item.caseData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.caseData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });
        searchResultsForDrafts = _.filter(readAllDrafts, function (item) {
          return item.draftData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.draftData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });  
      }
      this.setState({ searchResultsForCases, searchResultsForDrafts });
    } else {
    // else if (selectedSpecialty == '') {
    //   this.setState({
    //     searchResultsForCases: readAllCases,
    //     searchResultsForDrafts: readAllDrafts
    //   });
    // }
    //   else {
      this.setState({   
        searchResultsForCases: readAllCases,
        searchResultsForDrafts: readAllDrafts
      });
    }
  }

  componentDidMount() {
    const currentUserData = {
      currentUserId: this.props.userdata.id,
      currentUserRole: this.props.userdata.user_type,
      isDeleted: false,
      isCompleted: false
    }
    sendAmplitudeData('Open Inbox page', currentUserData)
    setTimeout(() => {
      this.props.fetchAllCases(currentUserData);
      this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
    }, 500)
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { readAllCases, readAllDrafts, errors } = nextProps;
    if (errors) {
      this.setState({
        errors: errors
      });
    }
    this.setState({
      readAllCases,
      readAllDrafts,
      searchResultsForCases: readAllCases,
      searchResultsForDrafts: readAllDrafts
    })
    if ((nextProps.isDraftDeleted !== this.props.isDraftDeleted) && nextProps.isDraftDeleted) {
      const currentUserData = {
        currentUserId: this.props.userdata.id,
        currentUserRole: this.props.userdata.user_type,
        isDeleted: false,
        isCompleted: false
      }
      setTimeout(() => {
        this.props.fetchAllCases(currentUserData);
        this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
      }, 500)
    }
  }
  removeDraft() {
    this.setState({ confirmDraftDelete: false })
    const { filesDraft, draftId } = this.state;
    if (filesDraft && filesDraft.length) {
      filesDraft.map(async (fileObj, index) => {
        if (fileObj && fileObj.name) {
          return await removeFileFromS3(fileObj, index);
        }
      })
    }
    this.props.removeDraft({ case_id: draftId })

  }
  handleClose() { this.setState({ confirmDraftDelete: false }) }

  handleSnackbarClose = () => { 
    this.props.setSnackbar(false);
  }

  openDraftDialog(draftData) {
    this.setState({ confirmDraftDelete: true, draftId: draftData.case_id, filesDraft: draftData.files && draftData.files.length ? draftData.files : [] })
  }

  handleAssignConsult = (id, specialty) => {
    if(id){
      this.setState({redirectToAssignConsult: true, cases_id: id, specialty: specialty});
    }
  }

  render() {
    const { classes, userdata, readAllDrafts } = this.props;
    const {
      redirectToChat,
      redirectToCreateCase,
      selectedClientID,
      selectedSpecialty,
      searchResultsForCases,
      searchResultsForDrafts,
      redirectToAssignConsult,
      cases_id,
      specialty
    } = this.state;
    if (redirectToChat) {
      return (<Redirect to={{
        pathname: `/insidemessage/room_${selectedClientID}`,
        state: { from: 'cases' }
      }} />)
    }
    if (redirectToCreateCase) {
      return (<Redirect to={{
        pathname: `/edit-new-case-${selectedClientID}`,
      }} />)
    }
    if(redirectToAssignConsult){
      return (<Redirect to={{
        pathname: `/assign-consult/case_${cases_id}/specialty_${specialty}`,
      }} />)
    }

    let rows = [...searchResultsForDrafts, ...searchResultsForCases];

    return (
      <div className={classes.container}>
        <CircularProgress isLoading={this.props.draftLoader} />
        <CircularProgress isLoading={this.props.caseLoader} />
        {/* 
        <div>
          <Media queries={{
            small: "(max-width: 599px)",
            medium: "(min-width: 600px) and (max-width: 1199px)",
            large: "(min-width: 1200px)"
          }}>
            {matches => (
              <Fragment>
                {matches.small && <p>I am small!</p> && <style>  </style>}
                {matches.medium && <p>I am medium!</p>}
                {matches.large && <p>I am large!</p>}
              </Fragment>
            )}
          </Media>
        </div> */}

        {/* <Hidden lgUp>
          <Grid container spacing={4} style={{ paddingBottom: 20 }}>
            <Grid item xs={5}
              style={{
                flexDirection: "row",
                display: "flex",
                justifyContent: "flex-start"
              }}
            >
              <Button
                color="primary"
                component={AdapterLink}
                to="/inbox/cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  width: 130,
                  // margin: 5
                }}>
                Open Cases
              </Button>
              <Button
                color="default"
                component={AdapterLink}
                to="/inbox/closed_cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  backgroundColor: "#ffffff",
                  width: 130,
                  marginLeft: 20,
                }}>
                Closed Cases
              </Button>
              {readAllDrafts && readAllDrafts.length ?
                <Button
                  color="default"
                  component={AdapterLink}
                  to="/inbox/drafts"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    backgroundColor: "#ffffff",
                    width: 80,
                    marginLeft: 32,
                  }}>
                  Drafts
              </Button> : null}
            </Grid>
            <Grid item xs={7}
              style={{
                display: "flex",
                justifyContent: "flex-end"
              }}
            >
              {userdata.user_type === "PCP" && (
                <Button
                  color="primary"
                  component={AdapterLink}
                  to="/create-new-case"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    marginLeft: 32,
                    height: 45
                  }}>
                  Create a new Case
              </Button>)}
            </Grid>
          </Grid>
        </Hidden> */}
        <Grid container justify="flex-start" spacing={0} className={classes.cardContainer} >
          <Grid item xs={12}>
            <Card style={{
              backgroundColor: "#ffffff",
              padding: 20,
              overflow: "auto",
              height: "100%"
            }}>
              <Grid  container justify="space-between" >
                <Grid style={{ paddingTop: 6, paddingLeft: 15, paddingRight: 15, paddingBottom: 20,
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"}}>
                  <Typography
                    color="textSecondary"
                    variant="h5"
                    fontWeight="100"
                  >
                    Open Cases
                  </Typography>
                  <Divider />
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4} 
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}>
                  <TextField
                    fullWidth
                    select
                    id="outlined-adornment-password"
                    className={classes.searchBar}
                    variant="outlined"
                    type={'text'}
                    size="small"
                    label="Filter by Specialty"
                    //label="Specialty type"
                    name={"selectedSpecialty"}
                    value={selectedSpecialty}
                    onChange={this.handleSearch}
                    style={{
                      display: userdata.user_type === "Admin" || userdata.user_type === "PCP"
                        ? "flex"
                        : "none"
                    }}
                  >
                    {specialtyList.map(specialty => (
                      <MenuItem key={specialty.value} value={specialty.value}>{specialty.label}</MenuItem>
                    ))}
                  </TextField>
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4}   
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}> 
                    <TextField
                      fullWidth
                      variant="outlined"
                      placeholder="Search"
                      size="small"
                      className={classes.searchBar}
                      InputProps={{
                       startAdornment: (
                         <InputAdornment position="start">
                           <SearchIcon  color="action"/>
                         </InputAdornment>
                       ),
                      }}
                      onChange={this.handleSearch}
                    />
                </Grid>
              </Grid>
              <MessageList
                openDraftDialog={this.openDraftDialog}
                rows={rows}
                isTrClicked={this.handleTrClicked}
                readAllCases={searchResultsForCases}
                readAllDrafts={searchResultsForDrafts}
                assignConsult={this.handleAssignConsult}
              />
            </Card>
            
          </Grid>
        </Grid>
        <ConfirmationDialog
          message="Are you sure you want to delete this draft?"
          title="Delete Draft"
          okOperationDialog={this.removeDraft}
          handleClose={this.handleClose}
          dialogState={this.state.confirmDraftDelete} />

          <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.props.showSnackbar}
            autoHideDuration={3000}
            onClose={this.handleSnackbarClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleSnackbarClose}
              variant="success"
              message="Case sent successfully."
            />
          </Snackbar>
      </div>
    );
  }
}

Inbox.propTypes = {
  userdata: PropTypes.object.isRequired,
  fetchAllCases: PropTypes.func.isRequired,
  setCurrentCase: PropTypes.func.isRequired,
  modifyCase: PropTypes.func.isRequired,
  fetchAllDrafts: PropTypes.func.isRequired,
  setCurrentDraft: PropTypes.func.isRequired,
  readAllCases: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata,
  readAllCases: state.case.readAllCases,
  readAllDrafts: state.draft.readAllDrafts,
  isDraftDeleted: state.draft.isDraftDeleted,
  draftLoader: state.draft.draftLoader,
  caseLoader: state.case.caseLoader,
  showSnackbar: state.case.showSnackbar
})

export default connect(
  mapStateToProps,
  { fetchAllCases, setCurrentCase, modifyCase, fetchAllDrafts, setCurrentDraft, removeDraft, setSnackbar }
)(withStyles(styles)(Inbox))