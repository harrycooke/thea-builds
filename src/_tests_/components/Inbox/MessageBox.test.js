import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import {MessageBox} from '../../../components/Inbox/MessageBox';

describe('MessageBox', () => {
    // beforeAll(() => {
    //     this.commonProps = {};
    // });
    

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(MessageBox, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
        const renderer = TestRenderer.create(MessageBox);
        renderer.toJSON();
      });

    describe('MessageBox', () => {
        it('should render correctly in "debug" mode', () => {
          const component = (MessageBox);
          expect(component).toMatchSnapshot();
        });
      });
});