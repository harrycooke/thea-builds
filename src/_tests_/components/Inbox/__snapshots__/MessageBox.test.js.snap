// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox MessageBox should render correctly in "debug" mode 1`] = `[Function]`;
